scripts/gwf

```diff
@@ -42,279 +38,188 @@ parser.add_argument('-e', '--pattern', default=False, action='store_true',
-        if len(unfinished_jobs) > 0:
-
-            print
-
-            max_state_length = 0
-            for _, state, _ in unfinished_jobs:
-                max_state_length = max(max_state_length, len(state))
-
-            for job_id, state, job_name in unfinished_jobs:
-                format_string = '{job_id:>8}  {state:<%d}  {job_name}' % max_state_length
-                print format_string.format(job_id=job_id, state=state, job_name=job_name)
-
-    sys.exit(0)
+working_dir = os.path.abspath(os.path.dirname(args.workflow_file))
 
 execfile(args.workflow_file)
-from gwf_workflow.workflow import build_workflow, schedule, dependencies
-workflow = build_workflow()
 
 if args.clean:
     if len(args.targets) == 0:
-        parser.error("No targets specified to clean.\nThis is never a good idea and gwf refuses to clean default targets.\n")
+        parser.error(
+            "No targets specified to clean.\nThis is never a good idea and gwf refuses to clean default targets.\n")
 
     if args.dry_run:
         for target in args.targets:
-            print 'Cleaning', target, 'will delete the following files:'
-            print workflow.targets[target].get_existing_outfiles()
-            print
+            print('Cleaning', target, 'will delete the following files:')
+            print(workflow[target].get_existing_outfiles())
+            print()
     else:
         for target in args.targets:
-            workflow.targets[target].clean_target()
+            workflow[target].clean_target()
 
     sys.exit(0)
 
 if len(args.targets) > 0:
     all_targets = args.targets
 else:
     # take all terminal nodes as default targets
-    all_targets = [n.target.name for n in workflow.targets.values() if len(n.dependents) == 0]
+    all_targets = [n.name for n in workflow.values() if len(n.dependents) == 0]
 
 if args.pattern:
-    defined_targets = [n.target.name for n in workflow.targets.values()]
+    defined_targets = [n.name for n in workflow.values()]
     all_targets = [fn
                    for pattern in all_targets
                    for fn in fnmatch.filter(defined_targets, pattern)]
 
 
-def split_tasks(tasks):
-    up_to_date, in_queue, to_schedule = [], [], []
-    for task in tasks:
-        if task.job_in_queue:
-            in_queue.append(task)
-        elif task.should_run:
-            to_schedule.append(task)
-        else:
-            up_to_date.append(task)
-    return up_to_date, in_queue, to_schedule
-
-
 if args.status:
     all_scheduled = set()
     for target in all_targets:
 
-        tasks = dependencies(workflow.targets, target)
+        tasks = dependencies(workflow, target)
         up_to_date, in_queue, to_schedule = split_tasks(tasks)
 
-        n_up_to_date, n_in_queue, n_to_schedule = len(up_to_date), len(in_queue), len(to_schedule)
+        n_up_to_date, n_in_queue, n_to_schedule = len(
+            up_to_date), len(in_queue), len(to_schedule)
```

Comment:
Agreed ;) The entire script has been deleted since it needs to be rewritten from scratch anyway using the new `Workflow` API.